From da41eb604df5b47df25c3b72070ff3b4d8dd55cd Mon Sep 17 00:00:00 2001
From: Usama Arif <usama.arif@arm.com>
Date: Fri, 8 Oct 2021 10:08:42 +0100
Subject: [PATCH 08/18] module: snps_umctl: Update DDR configuration and booker
 hashing

This is to address the speed difference between emulator and FPGA.
The hashing for RNSAM and HNF has also changed.

Signed-off-by: Usama Arif <usama.arif@arm.com>
---
 module/cmn_booker/src/mod_cmn_booker.c        |   4 +-
 .../include/DWC_ddrctl_lpddr5_init_mem.h      |  21 +-
 .../include/DWC_ddrctl_lpddr5_init_regs.h     |  57 +-
 module/snps_umctl/include/dmc620_registers.h  | 860 +++++++++---------
 module/snps_umctl/include/snps_defines.h      |  10 +
 module/snps_umctl/include/test_defs.h         |  15 +-
 6 files changed, 520 insertions(+), 447 deletions(-)

diff --git a/module/cmn_booker/src/mod_cmn_booker.c b/module/cmn_booker/src/mod_cmn_booker.c
index fb6aaafa..fb817338 100644
--- a/module/cmn_booker/src/mod_cmn_booker.c
+++ b/module/cmn_booker/src/mod_cmn_booker.c
@@ -93,7 +93,7 @@ static void process_node_hnf(struct cmn_booker_hnf_reg *hnf)
     hnf->SAM_CONTROL = config->snf_table[logical_id];
 
     /* Set hashing for HNF */
-    hnf->HASH_ADDR_MASK_REG = 0xFFFFFFC012F00;
+    hnf->HASH_ADDR_MASK_REG = 0xffffffffff00;
 
     /*
      * Map sub-regions to this HN-F node
@@ -402,7 +402,7 @@ static int cmn_booker_setup_sam(struct cmn_booker_rnsam_reg *rnsam)
                  CMN_BOOKER_RNSAM_NON_HASH_TGT_NODEID_ENTRY_MASK) << bit_pos;
 
             /* Set hashing for RNSAM */
-            rnsam->HASH_ADDR_MASK_REG = 0xFFFFFFC012F00;
+            rnsam->HASH_ADDR_MASK_REG = 0xffffffffff00;
 
             region_io_count++;
             break;
diff --git a/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_mem.h b/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_mem.h
index 253a986e..bac19f88 100644
--- a/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_mem.h
+++ b/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_mem.h
@@ -371,9 +371,28 @@ while ((MEM_RW(DMC_BASE, 0x010090)&0x1)  ==1 ); //MRSTAT
 
   while ((MEM_RW(DMC_BASE, 0x10014))  !=1 ) ;  // STAT - wait until normal
 
+#if DMC_WCK_ALWAYS_ON
+  //MEM_RW(DMC_BASE, 0x10180) = (MEM_RW(DMC_BASE, 0x10180)|(0x1<<11)); // PWRCTL - selfref_sw=1 
+  MEM_RW(DMC_BASE, 0x10180) = (0x1<<11); // PWRCTL - selfref_sw=1 
+  while (((MEM_RW(DMC_BASE, 0x10014)) & 0x00000037) !=0x23 ) ; // STAT - wait until STAT.operating_mode=3b011 and STAT.selfref_type=2b10.
+  MEM_RW(DMC_BASE, 0x10c80) = 0x00000000; // SWCTL -sw_done=0 Enable writes to quasi-dynamic registers outside reset.
+  MEM_RW(DMC_BASE, 0x10010) = 0x00000001; // MSTR4 - CH0 WCK Always on Mode Enable 
+  //MEM_RW(DMC_BASE, 0x11010) = 0x00000001; // MSTR4 - CH1 WCK Always on Mode Enable 
+  MEM_RW(DMC_BASE, 0x10c80) = 0x00000001; // SWCTL - sw_done =1 Enable writes to quasi-dynamic registers outside reset.
+  while ((MEM_RW(DMC_BASE, 0x10c84)) != 0x00000001); // SWSTAT-sw_done_ack=1 propagtes to ack
+  MEM_RW(DMC_BASE, 0x10180) = (0x0<<11); // PWRCTL - selfref_sw=1 
+  //((MEM_RW(DMC_BASE, 0x10180)) | (0x0<<11)); // PWRCTL - selfref_sw=0 
+  while (((MEM_RW(DMC_BASE, 0x10014)) & 0x00000007) !=0x1 ) ; // STAT - wait until STAT.operating_mode=3b011 and STAT.selfref_type=2b10.
+  //while ((MEM_RW(DMC_BASE, 0x10014))  !=1 ) ;  // STAT - wait until normal
+#endif
+
 //MRSTAT //MR2 WL =16 RL=17
 MEM_RW(DMC_BASE, 0x010080)  = 0x00000030; //MRCTRL0 - Program the Rank#
-MEM_RW(DMC_BASE, 0x010084)  = 0x0000026b; //MRCTRL1 - Program MR Address and MR WData
+#if DMC_WCK_ALWAYS_ON
+    MEM_RW(DMC_BASE, 0x010084)  = 0x0000026b; //MRCTRL1 - Program MR Address and MR WData
+#else
+    MEM_RW(DMC_BASE, 0x010084)  = 0x0000029b; //MRCTRL1 - Program MR Address and MR WData
+#endif
 MEM_RW(DMC_BASE, 0x010080)  = 0x80000030; //MRCTRL0 - Trigger MR Write from the controller
 while ((MEM_RW(DMC_BASE, 0x010090)&0x1)  ==1 );
 //
diff --git a/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_regs.h b/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_regs.h
index e834af80..1c9d5701 100644
--- a/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_regs.h
+++ b/module/snps_umctl/include/DWC_ddrctl_lpddr5_init_regs.h
@@ -56,7 +56,7 @@ MEM_RW(DMC_BASE, 0x010d04)  = 0x00010000; //INITTMG1 - Number of cycles for SDRA
 // AXI(40Bits) - System Address            | 0  | 1  | 2  | 3  | 4  | 5  | 6  | 7  | 8  | 9  | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 | 28 | 29 | 30 | 31 | 32 | 33 | 34 | 35 | 36 | 37 | 38 | 39 |
 // ----------------------------------------+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
 // Below registers are used to map the HIF address to SDRAM address
-MEM_RW(DMC_BASE, 0x030004)  = 0x3f3f3f08; //ADDRMAP1  (addrmap_cs_b3_0)         ->  addrmap_cs_b3 = 0x3f(unused), addrmap_cs_b2 = 0x3f(unused), addrmap_cs_b1 = 0x3f(unused), addrmap_cs_b0 = (14[HIF] - 6[base]) = 0x8
+MEM_RW(DMC_BASE, 0x030004)  = 0x3f3f1f08; //ADDRMAP1  (addrmap_cs_b3_0)         ->  addrmap_cs_b3 = 0x3f(unused), addrmap_cs_b2 = 0x3f(unused), addrmap_cs_b1 = 0x3f(unused), addrmap_cs_b0 = (14[HIF] - 6[base]) = 0x8
 MEM_RW(DMC_BASE, 0x03000c)  = 0x3f3f0707; //ADDRMAP3  (addrmap_bank_b2_0)       ->  addrmap_bank_b2 = 0x3f(unused), addrmap_bank_b1 = (11[HIF] - 4[base]) = 0x07, addrmap_bank_b0 = (10[HIF] - 3[base]) = 0x07
 MEM_RW(DMC_BASE, 0x030010)  = 0x003f0909; //ADDRMAP4  (addrmap_bank_bg_b2_0)    ->  addrmap_bank_bg_b2 = unused, addrmap_bank_bg_b1 = (13[HIF] - 4[base]) = 0x09, addrmap_bank_bg_b0 = (12[HIF] - 3[base]) = 0x09 
 MEM_RW(DMC_BASE, 0x030014)  = 0x1f000000; //ADDRMAP5  (addrmap_col_b10_7)       ->  addrmap_col_b10 = 0x1f(unused), addrmap_col_b9 = (9[HIF] - 9[base]) = 0x00, addrmap_col_b8 = (8[HIF] - 8[base]) = 0x00, addrmap_col_b7 = (7[HIF] - 7[base]) = 0x00
@@ -193,8 +193,10 @@ MEM_RW(DMC_BASE, 0x10390) = (8<<24) | (0x40<<16) | (UMCTL_WR_ACT_GAP << 8)
   
 // Below registers are used to map the HIF address to SDRAM address
 MEM_RW(DMC_BASE, 0x030004)  = 0x3f3f3f08; //ADDRMAP1  (addrmap_cs_b3_0)         ->  addrmap_cs_b3 = 0x3f(unused), addrmap_cs_b2 = 0x3f(unused), addrmap_cs_b1 = 0x3f(unused), addrmap_cs_b0 = (14[HIF] - 6[base]) = 0x8
-MEM_RW(DMC_BASE, 0x03000c)  = 0x3f3f0707; //ADDRMAP3  (addrmap_bank_b2_0)       ->  addrmap_bank_b2 = 0x3f(unused), addrmap_bank_b1 = (11[HIF] - 4[base]) = 0x07, addrmap_bank_b0 = (10[HIF] - 3[base]) = 0x07
-MEM_RW(DMC_BASE, 0x030010)  = 0x003f0909; //ADDRMAP4  (addrmap_bank_bg_b2_0)    ->  addrmap_bank_bg_b2 = unused, addrmap_bank_bg_b1 = (13[HIF] - 4[base]) = 0x09, addrmap_bank_bg_b0 = (12[HIF] - 3[base]) = 0x09 
+//MEM_RW(DMC_BASE, 0x03000c)  = 0x3f3f0707; //ADDRMAP3  (addrmap_bank_b2_0)       ->  addrmap_bank_b2 = 0x3f(unused), addrmap_bank_b1 = (11[HIF] - 4[base]) = 0x07, addrmap_bank_b0 = (10[HIF] - 3[base]) = 0x07
+//MEM_RW(DMC_BASE, 0x030010)  = 0x003f0909; //ADDRMAP4  (addrmap_bank_bg_b2_0)    ->  addrmap_bank_bg_b2 = unused, addrmap_bank_bg_b1 = (13[HIF] - 4[base]) = 0x09, addrmap_bank_bg_b0 = (12[HIF] - 3[base]) = 0x09 
+MEM_RW(DMC_BASE, 0x03000c)  = 0x003f0909; //ADDRMAP3  (addrmap_bank_b2_0)       ->  addrmap_bank_b2 = 0x3f(unused), addrmap_bank_b1 = (13[HIF] - 4[base]) = 0x09, addrmap_bank_b0 = (12[HIF] - 3[base]) = 0x09
+MEM_RW(DMC_BASE, 0x030010)  = 0x003f0707; //ADDRMAP4  (addrmap_bank_bg_b2_0)    ->  addrmap_bank_bg_b2 = unused, addrmap_bank_bg_b1 = (11[HIF] - 4[base]) = 0x07, addrmap_bank_bg_b0 = (10[HIF] - 3[base]) = 0x07
 MEM_RW(DMC_BASE, 0x030014)  = 0x1f000000; //ADDRMAP5  (addrmap_col_b10_7)       ->  addrmap_col_b10 = 0x1f(unused), addrmap_col_b9 = (9[HIF] - 9[base]) = 0x00, addrmap_col_b8 = (8[HIF] - 8[base]) = 0x00, addrmap_col_b7 = (7[HIF] - 7[base]) = 0x00
 MEM_RW(DMC_BASE, 0x030018)  = 0x00000000; //ADDRMAP6  (addrmap_col_b6_3)        ->  addrmap_col_b6 = (6[HIF] - 6[base]) = 0x00, addrmap_col_b5 = (5[HIF] - 5[base]) = 0x00, addrmap_col_b4 = (5[HIF] - 5[base]) = 0x00, addrmap_col_b4 = (4[HIF] - 4[base]) = 0x00
 MEM_RW(DMC_BASE, 0x03001c)  = 0x1f1f1f09; //ADDRMAP7  (addrmap_row_b17_14)      ->  addrmap_row_b17 = 0x1f(unused), addrmap_row_b16 = 0x1f(unused), addrmap_row_b15 = 0x1f(unused), addrmap_row_b14 = (29[HIF] - 20[base]) = 0x09  
@@ -206,30 +208,31 @@ MEM_RW(DMC_BASE, 0x030030)  = 0x00000000; //ADDRMAP12 (nonbinary_device_density)
 #ifdef DMC_DISABLE_REFRESH
   MEM_RW(DMC_BASE, 0x010208) = 0x00000001; // RFSHCTL0  - dis_auto_ref
 #endif
-MEM_RW(DMC_BASE, 0x000d04)  = 0x00000e12; //RANKTMG0
-MEM_RW(DMC_BASE, 0x000d08)  = 0x00000f11; //RANKTMG1
-MEM_RW(DMC_BASE, 0x000000)  = 0x2f101b22; //DRAMSET1TMG0
-MEM_RW(DMC_BASE, 0x000004)  = 0x00060633; //DRAMSET1TMG1
-MEM_RW(DMC_BASE, 0x000008)  = 0x10110a1e; //DRAMSET1TMG2 - WL 16 RL - 17
-MEM_RW(DMC_BASE, 0x00000c)  = 0x000c2136; //DRAMSET1TMG3
-MEM_RW(DMC_BASE, 0x000010)  = 0x0f040412; //DRAMSET1TMG4 - t_rcd = 15 t_ccd_l = 4 t_rrd_l = 4 t_rp = 18
-MEM_RW(DMC_BASE, 0x000014)  = 0x02040c01; //DRAMSET1TMG5
-MEM_RW(DMC_BASE, 0x000018)  = 0x00000008; //DRAMSET1TMG6
-MEM_RW(DMC_BASE, 0x00001c)  = 0x00000003; //DRAMSET1TMG7
-MEM_RW(DMC_BASE, 0x000020)  = 0x00004400; //DRAMSET1TMG8
-MEM_RW(DMC_BASE, 0x000024)  = 0x00020419; //DRAMSET1TMG9 - tccd_s = 2 t_rrd_s = 4 twr2rd_s = 27
-MEM_RW(DMC_BASE, 0x000030)  = 0x00030000; //DRAMSET1TMG12
-MEM_RW(DMC_BASE, 0x000034)  = 0x13100002; //DRAMSET1TMG13
-MEM_RW(DMC_BASE, 0x000038)  = 0x0020006e; //DRAMSET1TMG14
-MEM_RW(DMC_BASE, 0x00005c)  = 0x009d05cf; //DRAMSET1TMG23
-MEM_RW(DMC_BASE, 0x000060)  = 0x00081615; //DRAMSET1TMG24 - 4Bank/4BG 
-MEM_RW(DMC_BASE, 0x000064)  = 0x00002f06; //DRAMSET1TMG25
-MEM_RW(DMC_BASE, 0x000078)  = 0x00190b18; //DRAMSET1TMG30
-MEM_RW(DMC_BASE, 0x000580)  = 0x0627002f; //DFITMG0 - tctldelay = 6 phy_dfi_clk = 1 trddataen = 39 tphywrdata = 0 tphywrlat = 64
-MEM_RW(DMC_BASE, 0x000584)  = 0x00140606; //DFITMG1
-MEM_RW(DMC_BASE, 0x000588)  = 0x00243f3b; //DFITMG2
-MEM_RW(DMC_BASE, 0x000590)  = 0x1C2C0400; //DFITMG4
-MEM_RW(DMC_BASE, 0x000594)  = 0x0410000C; //DFITMG5
+MEM_RW(DMC_BASE, 0x00600)  = 0x82100186; //RFSHSET1TMG0 - t_refi_x1_sel = 1 refresh_margin=2 refresh_to_x1_x32=16 t_refi_x1_x32=390
+MEM_RW(DMC_BASE, 0x00d04)  = 0x00000b07; //RANKTMG0
+MEM_RW(DMC_BASE, 0x00d08)  = 0x00001102; //RANKTMG1
+MEM_RW(DMC_BASE, 0x00000)  = 0x28100322; //DRAMSET1TMG0
+MEM_RW(DMC_BASE, 0x00004)  = 0x00060630; //DRAMSET1TMG1
+MEM_RW(DMC_BASE, 0x00008)  = 0x09111117; //DRAMSET1TMG2 - WL 9 RL - 17
+MEM_RW(DMC_BASE, 0x0000c)  = 0x000c212f; //DRAMSET1TMG3
+MEM_RW(DMC_BASE, 0x00010)  = 0x0f04040f; //DRAMSET1TMG4 - t_rcd = 15 t_ccd_l = 4 t_rrd_l = 4 t_rp = 15
+MEM_RW(DMC_BASE, 0x00014)  = 0x02040c01; //DRAMSET1TMG5
+MEM_RW(DMC_BASE, 0x00018)  = 0x00000008; //DRAMSET1TMG6
+MEM_RW(DMC_BASE, 0x0001c)  = 0x00000003; //DRAMSET1TMG7
+MEM_RW(DMC_BASE, 0x00020)  = 0x00004400; //DRAMSET1TMG8
+MEM_RW(DMC_BASE, 0x00024)  = 0x00020412; //DRAMSET1TMG9 - tccd_s = 2 t_rrd_s = 4 twr2rd_s = 18
+MEM_RW(DMC_BASE, 0x00030)  = 0x00030000; //DRAMSET1TMG12
+MEM_RW(DMC_BASE, 0x00034)  = 0x0c100002; //DRAMSET1TMG13  odtloff=12 t_ccd_mw=16 t_ppd=2 for LPDDR4 only
+MEM_RW(DMC_BASE, 0x00038)  = 0x000000e6; //DRAMSET1TMG14
+MEM_RW(DMC_BASE, 0x0005c)  = 0x009d0bcc; //DRAMSET1TMG23
+MEM_RW(DMC_BASE, 0x00060)  = 0x000f160e; //DRAMSET1TMG24 - 4Bank/4BG
+MEM_RW(DMC_BASE, 0x00064)  = 0x00002806; //DRAMSET1TMG25
+MEM_RW(DMC_BASE, 0x00078)  = 0x00191218; //DRAMSET1TMG30
+MEM_RW(DMC_BASE, 0x00580)  = 0x0627002f; //DFITMG0 - tctldelay = 6 trddataen = 39 tphywrdata = 0 tphywrlat =47 // old latency pass
+MEM_RW(DMC_BASE, 0x00584)  = 0x00140606; //DFITMG1
+MEM_RW(DMC_BASE, 0x00588)  = 0x00243f1f; //DFITMG2
+MEM_RW(DMC_BASE, 0x00590)  = 0x180c0402; //DFITMG4
+MEM_RW(DMC_BASE, 0x00594)  = 0x04100006; //DFITMG5
 
   // Enable QOS
 #ifdef QOS_DISABLE
diff --git a/module/snps_umctl/include/dmc620_registers.h b/module/snps_umctl/include/dmc620_registers.h
index 88735a32..1c5656b5 100644
--- a/module/snps_umctl/include/dmc620_registers.h
+++ b/module/snps_umctl/include/dmc620_registers.h
@@ -1,425 +1,455 @@
+/* -----------------------------------------------------------------------------
+ * The confidential and proprietary information contained in this file may
+ * only be used by a person authorised under and to the extent permitted
+ * by a subsisting licensing agreement from ARM Limited.
+ *
+ *               (C) COPYRIGHT 2012 ARM Limited.
+ *                   ALL RIGHTS RESERVED
+ *
+ * This entire notice must be reproduced on all copies of this file
+ * and copies of this file may only be made by a person if such person is
+ * permitted to do so under the terms of a subsisting license agreement
+ * from ARM Limited.
+ *
+ * SVN Information
+ *
+ * Checked In : $Date: 2012-08-28 10:34:01 +0100 (Tue, 28 Aug 2012) $
+ * Revision : $Revision: 38152 $
+ *
+ * Release Information : $State: $
+ *
+ * -----------------------------------------------------------------------------
+ * Purpose : Register description for DMC620
+ *
+ * -----------------------------------------------------------------------------
+ */
 
-#define FPGAPHY_GENERAL_SETUP     					0x0000
-#define FPGAPHY_PLAYBACK_ENABLE     				0x0004
-#define FPGAPHY_PLAYBACK_STATUS     				0x0008
-#define FPGAPHY_WRITE_PIPE_DELAY     				0x000C
-#define FPGAPHY_CA_PIPE_DELAY     					0x0010
-#define FPGAPHY_DATA_BYTE_SEL     					0x0014
-#define FPGAPHY_CTRLUPD_CONFIG     					0x0018
-#define FPGAPHY_DEBUG     							0x001C
-#define FPGAPHY_INTERRUPT_MASK     					0x0020
-#define FPGAPHY_INTERRUPT_ENABLE     				0x0024
-#define FPGAPHY_INTERRUPT_STATUS     				0x0028
-#define FPGAPHY_INTERRUPT_CLEAR     				0x002C
-#define FPGAPHY_DFICLK_READ_DELAY_RANK_0     		0x0030
-#define FPGAPHY_FCLK_READ_DELAY_RANK_0_BYTES_7_0    0x0034
-#define FPGAPHY_FCLK_READ_DELAY_RANK_0_BYTE_8     	0x0038
-#define FPGAPHY_DFICLK_READ_DELAY_RANK_1     		0x0040
-#define FPGAPHY_FCLK_READ_DELAY_RANK_1_BYTES_7_0    0x0044
-#define FPGAPHY_FCLK_READ_DELAY_RANK_1_BYTE_8     	0x0048
-#define FPGAPHY_FCLK_READ_OFFSET_RANK_0_BYTES_7_0   0x0050
-#define FPGAPHY_FCLK_READ_OFFSET_RANK_0_BYTE_8     	0x0054
-#define FPGAPHY_FCLK_READ_OFFSET_RANK_1_BYTES_7_0   0x0058
-#define FPGAPHY_FCLK_READ_OFFSET_RANK_1_BYTE_8      0x005C
-#define FPGAPHY_AUTOTRAIN_LOOP_ADDR     			0x0060
-#define FPGAPHY_TRAIN_STATUS     					0x0064
-#define FPGAPHY_READ_TRAINING_CAPTURE_0     		0x0068
-#define FPGAPHY_READ_TRAINING_CAPTURE_1     		0x006C
-#define FPGAPHY_REVISION_REGISTER     				0x0070
-#define FPGAPHY_READ_PHASE_DELAY_RANK_0     		0x0074
-#define FPGAPHY_READ_PHASE_DELAY_RANK_1     		0x0078
-#define PLAYBACK_ENGINE_CODE_0						0x0800
-#define PLAYBACK_ENGINE_CODE_1						0x0804
-#define PLAYBACK_ENGINE_CODE_2						0x0808
-#define PLAYBACK_ENGINE_CODE_3						0x080C
-#define PLAYBACK_ENGINE_CODE_4						0x0810
-#define PLAYBACK_ENGINE_CODE_8 						0x0820
-#define PLAYBACK_ENGINE_CODE_9 						0x0824
-#define PLAYBACK_ENGINE_CODE_10 					0x0828
-#define PLAYBACK_ENGINE_CODE_11 					0x082c
-#define PLAYBACK_ENGINE_CODE_12 					0x0830
-#define PLAYBACK_ENGINE_CODE_16 					0x0840
-#define PLAYBACK_ENGINE_CODE_17 					0x0844
-#define PLAYBACK_ENGINE_CODE_18 					0x0848
-#define PLAYBACK_ENGINE_CODE_19 					0x084c
-#define PLAYBACK_ENGINE_CODE_20 					0x0850
-#define PLAYBACK_ENGINE_CODE_24 					0x0860
-#define PLAYBACK_ENGINE_CODE_25 					0x0864
-#define PLAYBACK_ENGINE_CODE_26 					0x0868
-#define PLAYBACK_ENGINE_CODE_27 					0x086c
-#define PLAYBACK_ENGINE_CODE_28 					0x0870
-#define PLAYBACK_ENGINE_CODE_32 					0x0880
-#define PLAYBACK_ENGINE_CODE_33 					0x0884
-#define PLAYBACK_ENGINE_CODE_34 					0x0888
-#define PLAYBACK_ENGINE_CODE_35 					0x088c
-#define PLAYBACK_ENGINE_CODE_36 					0x0890
-#define PLAYBACK_ENGINE_CODE_40 					0x08a0
-#define PLAYBACK_ENGINE_CODE_41 					0x08a4
-#define PLAYBACK_ENGINE_CODE_42 					0x08a8
-#define PLAYBACK_ENGINE_CODE_43 					0x08ac
-#define PLAYBACK_ENGINE_CODE_44 					0x08b0
-#define PLAYBACK_ENGINE_CODE_48 					0x08c0
-#define PLAYBACK_ENGINE_CODE_49 					0x08c4
-#define PLAYBACK_ENGINE_CODE_50 					0x08c8
-#define PLAYBACK_ENGINE_CODE_51 					0x08cc
-#define PLAYBACK_ENGINE_CODE_52 					0x08d0
-#define PLAYBACK_ENGINE_CODE_56 					0x08e0
-#define PLAYBACK_ENGINE_CODE_57 					0x08e4
-#define PLAYBACK_ENGINE_CODE_58 					0x08e8
-#define PLAYBACK_ENGINE_CODE_59 					0x08ec
-#define PLAYBACK_ENGINE_CODE_60 					0x08f0
-#define PLAYBACK_ENGINE_CODE_64 					0x0900
-#define PLAYBACK_ENGINE_CODE_65 					0x0904
-#define PLAYBACK_ENGINE_CODE_66 					0x0908
-#define PLAYBACK_ENGINE_CODE_67 					0x090c
-#define PLAYBACK_ENGINE_CODE_68 					0x0910
-#define PLAYBACK_ENGINE_CODE_72 					0x0920
-#define PLAYBACK_ENGINE_CODE_73 					0x0924
-#define PLAYBACK_ENGINE_CODE_74 					0x0928
-#define PLAYBACK_ENGINE_CODE_75 					0x092c
-#define PLAYBACK_ENGINE_CODE_76 					0x0930
-#define PLAYBACK_ENGINE_CODE_80 					0x0940
-#define PLAYBACK_ENGINE_CODE_81 					0x0944
-#define PLAYBACK_ENGINE_CODE_82 					0x0948
-#define PLAYBACK_ENGINE_CODE_83 					0x094c
-#define PLAYBACK_ENGINE_CODE_84 					0x0950
-#define PLAYBACK_ENGINE_CODE_88 					0x0960
-#define PLAYBACK_ENGINE_CODE_89 					0x0964
-#define PLAYBACK_ENGINE_CODE_90 					0x0968
-#define PLAYBACK_ENGINE_CODE_91 					0x096c
-#define PLAYBACK_ENGINE_CODE_92 					0x0970
-#define PLAYBACK_ENGINE_CODE_96 					0x0980
-#define PLAYBACK_ENGINE_CODE_97 					0x0984
-#define PLAYBACK_ENGINE_CODE_98 					0x0988
-#define PLAYBACK_ENGINE_CODE_99 					0x098c
-#define PLAYBACK_ENGINE_CODE_100 					0x0990
-#define PLAYBACK_ENGINE_CODE_104 					0x09a0
-#define PLAYBACK_ENGINE_CODE_105 					0x09a4
-#define PLAYBACK_ENGINE_CODE_106 					0x09a8
-#define PLAYBACK_ENGINE_CODE_107 					0x09ac
-#define PLAYBACK_ENGINE_CODE_108 					0x09b0
-#define PLAYBACK_ENGINE_CODE_112 					0x09c0
-#define PLAYBACK_ENGINE_CODE_113 					0x09c4
-#define PLAYBACK_ENGINE_CODE_114 					0x09c8
-#define PLAYBACK_ENGINE_CODE_115 					0x09cc
-#define PLAYBACK_ENGINE_CODE_116 					0x09d0
-#define PLAYBACK_ENGINE_CODE_120 					0x09e0
-#define PLAYBACK_ENGINE_CODE_121 					0x09e4
-#define PLAYBACK_ENGINE_CODE_122 					0x09e8
-#define PLAYBACK_ENGINE_CODE_123 					0x09ec
-#define PLAYBACK_ENGINE_CODE_124 					0x09f0
-#define PLAYBACK_ENGINE_CODE_128 					0x0a00
-#define PLAYBACK_ENGINE_CODE_129 					0x0a04
-#define PLAYBACK_ENGINE_CODE_130 					0x0a08
-#define PLAYBACK_ENGINE_CODE_131 					0x0a0c
-#define PLAYBACK_ENGINE_CODE_132 					0x0a10
-#define PLAYBACK_ENGINE_CODE_136 					0x0a20
-#define PLAYBACK_ENGINE_CODE_137 					0x0a24
-#define PLAYBACK_ENGINE_CODE_138 					0x0a28
-#define PLAYBACK_ENGINE_CODE_139 					0x0a2c
-#define PLAYBACK_ENGINE_CODE_140 					0x0a30
-#define PLAYBACK_ENGINE_CODE_144 					0x0a40
-#define PLAYBACK_ENGINE_CODE_145 					0x0a44
-#define PLAYBACK_ENGINE_CODE_146 					0x0a48
-#define PLAYBACK_ENGINE_CODE_147 					0x0a4c
-#define PLAYBACK_ENGINE_CODE_148 					0x0a50
-#define PLAYBACK_ENGINE_CODE_152 					0x0a60
-#define PLAYBACK_ENGINE_CODE_153 					0x0a64
-#define PLAYBACK_ENGINE_CODE_154 					0x0a68
-#define PLAYBACK_ENGINE_CODE_155 					0x0a6c
-#define PLAYBACK_ENGINE_CODE_156 					0x0a70
-#define PLAYBACK_ENGINE_CODE_160 					0x0a80
-#define PLAYBACK_ENGINE_CODE_161 					0x0a84
-#define PLAYBACK_ENGINE_CODE_162 					0x0a88
-#define PLAYBACK_ENGINE_CODE_163 					0x0a8c
-#define PLAYBACK_ENGINE_CODE_164 					0x0a90
-#define PLAYBACK_ENGINE_CODE_168 					0x0aa0
-#define PLAYBACK_ENGINE_CODE_169 					0x0aa4
-#define PLAYBACK_ENGINE_CODE_170 					0x0aa8
-#define PLAYBACK_ENGINE_CODE_171 					0x0aac
-#define PLAYBACK_ENGINE_CODE_172 					0x0ab0
-#define PLAYBACK_ENGINE_CODE_176 					0x0ac0
-#define PLAYBACK_ENGINE_CODE_177 					0x0ac4
-#define PLAYBACK_ENGINE_CODE_178 					0x0ac8
-#define PLAYBACK_ENGINE_CODE_179 					0x0acc
-#define PLAYBACK_ENGINE_CODE_180  					0x0ad0
-#define PLAYBACK_ENGINE_CODE_184  					0x0ae0
-#define PLAYBACK_ENGINE_CODE_185  					0x0ae4
-#define PLAYBACK_ENGINE_CODE_186  					0x0ae8
-#define PLAYBACK_ENGINE_CODE_187  					0x0aec
-#define PLAYBACK_ENGINE_CODE_188  					0x0af0
-#define PLAYBACK_ENGINE_CODE_192  					0x0b00
-#define PLAYBACK_ENGINE_CODE_193  					0x0b04
-#define PLAYBACK_ENGINE_CODE_194  					0x0b08
-#define PLAYBACK_ENGINE_CODE_195  					0x0b0c
-#define PLAYBACK_ENGINE_CODE_196  					0x0b10
-#define PLAYBACK_ENGINE_CODE_200  					0x0b20
-#define PLAYBACK_ENGINE_CODE_201  					0x0b24
-#define PLAYBACK_ENGINE_CODE_202  					0x0b28
-#define PLAYBACK_ENGINE_CODE_203  					0x0b2c
-#define PLAYBACK_ENGINE_CODE_204  					0x0b30
-#define PLAYBACK_ENGINE_CODE_208  					0x0b40
-#define PLAYBACK_ENGINE_CODE_209  					0x0b44
-#define PLAYBACK_ENGINE_CODE_210  					0x0b48
-#define PLAYBACK_ENGINE_CODE_211  					0x0b4c
-#define PLAYBACK_ENGINE_CODE_212  					0x0b50
-#define PLAYBACK_ENGINE_CODE_216  					0x0b60
-#define PLAYBACK_ENGINE_CODE_217  					0x0b64
-#define PLAYBACK_ENGINE_CODE_218  					0x0b68
-#define PLAYBACK_ENGINE_CODE_219  					0x0b6c
-#define PLAYBACK_ENGINE_CODE_220  					0x0b70
-#define PLAYBACK_ENGINE_CODE_224  					0x0b80
-#define PLAYBACK_ENGINE_CODE_225  					0x0b84
-#define PLAYBACK_ENGINE_CODE_226  					0x0b88
-#define PLAYBACK_ENGINE_CODE_227  					0x0b8c
-#define PLAYBACK_ENGINE_CODE_228  					0x0b90
-#define PLAYBACK_ENGINE_CODE_232  					0x0ba0
-#define PLAYBACK_ENGINE_CODE_233  					0x0ba4
-#define PLAYBACK_ENGINE_CODE_234  					0x0ba8
-#define PLAYBACK_ENGINE_CODE_235  					0x0bac
-#define PLAYBACK_ENGINE_CODE_236  					0x0bb0
-#define PLAYBACK_ENGINE_CODE_240  					0x0bc0
-#define PLAYBACK_ENGINE_CODE_241  					0x0bc4
-#define PLAYBACK_ENGINE_CODE_242  					0x0bc8
-#define PLAYBACK_ENGINE_CODE_243  					0x0bcc
-#define PLAYBACK_ENGINE_CODE_244  					0x0bd0
-#define PLAYBACK_ENGINE_CODE_248  					0x0be0
-#define PLAYBACK_ENGINE_CODE_249  					0x0be4
-#define PLAYBACK_ENGINE_CODE_250  					0x0be8
-#define PLAYBACK_ENGINE_CODE_251  					0x0bec
-#define PLAYBACK_ENGINE_CODE_252  					0x0bf0
-#define PLAYBACK_ENGINE_CODE_256  					0x0c00
-#define PLAYBACK_ENGINE_CODE_257  					0x0c04
-#define PLAYBACK_ENGINE_CODE_258  					0x0c08
-#define PLAYBACK_ENGINE_CODE_259  					0x0c0c
-#define PLAYBACK_ENGINE_CODE_260  					0x0c10
-#define PLAYBACK_ENGINE_CODE_264  					0x0c20
-#define PLAYBACK_ENGINE_CODE_265  					0x0c24
-#define PLAYBACK_ENGINE_CODE_266  					0x0c28
-#define PLAYBACK_ENGINE_CODE_267  					0x0c2c
-#define PLAYBACK_ENGINE_CODE_268  					0x0c30
- #define     DFIMISC  0x01b0
- #define     DFITMG0  0x0190
- #define     DFITMG1  0x0194
- #define   DFILPCFG0  0x0198
- #define     DFIUPD0  0x01a0
- #define     DFIUPD1  0x01a4
- #define     DFIUPD2  0x01a8
+/* Register definitions for PHY */
+#ifndef __DMC620_REG_H__
+#define __DMC620_REG_H__ 
 
-#define STAT 	0x0004
-#define SWSTAT 	0x0324
+#define DMC620_PHY_INIT_COMPLETE                            0x000
+#define DMC620_PHY_READ_DELAY                               0x008
+#define DMC620_PHY_CAPTURE_MASK                             0x00C
+#define DMC620_PHY_T_CONTROL_DELAY                          0x014
+#define DMC620_PHY_WR_LATENCY                               0x018
+#define DMC620_PHY_RDDATA_EN                                0x01c
+#define DMC620_PHY_RD_LATENCY                               0x020
+#define DMC620_PHY_CLP_ACK                                  0x030
+#define DMC620_PHY_RDLVL_REQ                                0x044
+#define DMC620_PHY_WRLVL_REQ                                0x04C
+#define DMC620_PHY_WR_PREMBL                                0x058
+#define DMC620_PHY_DELAY_SEL				                        0x820
+#define DMC620_PHY_REF_EN				                            0x824
+#define DMC620_PHY_T_CTRL_UPD_MIN                           0x828
+#define DMC620_PHY_OP_DELAY1                                0x830
+#define DMC620_PHY_OP_DELAY2                                0x834
+#define DMC620_PHY_OP_DELAY3                                0x838
+#define DMC620_PHY_OP_DELAY4                                0x83C
 
- #define         DBG1  	0x0304
- #define       PWRCTL  	0x0030
- #define         MSTR  	0x0000
- #define 		MRSTAT  0x0018
- #define        INIT0  	0x00d0
- #define        INIT1  	0x00d4
- #define        INIT2  	0x00d8
- #define        INIT3  	0x00dc
- #define        INIT4  	0x00e0
- #define        INIT5  	0x00e4
- #define        INIT6  	0x00e8
- #define	  RANKCTL  	0x00f4
- #define     DRAMTMG0  	0x0100
- #define     DRAMTMG1  	0x0104
- #define     DRAMTMG2  	0x0108
- #define     DRAMTMG3  	0x010c
- #define     DRAMTMG4  	0x0110
- #define     DRAMTMG5  	0x0114
- #define     DRAMTMG6  	0x0118
- #define     DRAMTMG7  	0x011c
- #define     DRAMTMG13  0x0134
- #define     DRAMTMG14  0x0138
- #define     ADDRMAP0  	0x0200
- #define     ADDRMAP1  	0x0204
- #define     ADDRMAP2  	0x0208
- #define     ADDRMAP3  	0x020c
- #define     ADDRMAP4  	0x0210
- #define     ADDRMAP5  	0x0214
- #define     ADDRMAP6  	0x0218
- #define 	ADDRMAP7 	0x021c
- #define 	ADDRMAP8 	0x0220
- #define     ADDRMAP9  	0x0224
- #define    ADDRMAP10  	0x0228
- #define    ADDRMAP11 	0x022c
+#define DMC620_STATUS					                              0x000
+#define DMC620_CONFIG					                              0x004
+#define DMC620_COMMAND				                        	    0x008
+#define DMC620_ADD_CONTROL		                       		    0x010
+#define DMC620_DEC_CONTROL		                       		    0x014
+#define DMC620_FORMAT_CONTROL	                     			    0x018
+#define DMC620_ADDRESS_MAP				                          0x01c
+#define DMC620_LOW_PWR_CONTROL				                      0x020
+#define DMC620_TURNAROUND_CONTROL			                      0x028
+#define DMC620_HIT_TURNAROUND_CONTROL			                  0x02c
+#define DMC620_QOS_CLASS_CONTROL			                      0x030
+#define DMC620_ESCALATION_CONTROL			                      0x034
+#define DMC620_QV_CONTROL_31_00				                      0x038
+#define DMC620_QV_CONTROL_63_32				                      0x03c
+#define DMC620_RT_CONTROL_31_00				                      0x040
+#define DMC620_RT_CONTROL_63_32				                      0x044
+#define DMC620_TIMEOUT_CONTROL				                      0x048
+#define DMC620_CREDIT_CONTROL				                        0x04c
+#define DMC620_WRITE_PRIORITY_CONTROL_31_00		              0x050
+#define DMC620_WRITE_PRIORITY_CONTROL_63_32		              0x054
+#define DMC620_QUEUE_THRESHOLD_CONTROL_31_00                0x058
+#define DMC620_QUEUE_THRESHOLD_CONTROL_63_32                0x05c
+#define DMC620_ADDR_SHUTTER_31_00			                      0x060
+#define DMC620_ADDR_SHUTTER_63_32			                      0x064
+#define DMC620_ADDR_SHUTTER_95_64			                      0x068
+#define DMC620_ADDR_SHUTTER_127_96			                    0x06c
+#define DMC620_ADDR_SHUTTER_159_128			                    0x070
+#define DMC620_ADDR_SHUTTER_191_160			                    0x074
+#define DMC620_MEMORY_ADDRESS_MAX_31_00			                0x078
+#define DMC620_MEMORY_ADDRESS_MAX_43_32			                0x07C
+#define DMC620_ACCESS_ADDRESS_MIN0_31_00		                0x080
+#define DMC620_ACCESS_ADDRESS_MIN0_43_32		                0x084
+#define DMC620_ACCESS_ADDRESS_MAX0_31_00		                0x088
+#define DMC620_ACCESS_ADDRESS_MAX0_43_32		                0x08C
+#define DMC620_ACCESS_ADDRESS_MIN1_31_00		                0x090
+#define DMC620_ACCESS_ADDRESS_MIN1_43_32		                0x094
+#define DMC620_ACCESS_ADDRESS_MAX1_31_00		                0x098
+#define DMC620_ACCESS_ADDRESS_MAX1_43_32		                0x09C
+#define DMC620_ACCESS_ADDRESS_MIN2_31_00		                0x0a0
+#define DMC620_ACCESS_ADDRESS_MIN2_43_32		                0x0a4
+#define DMC620_ACCESS_ADDRESS_MAX2_31_00		                0x0a8
+#define DMC620_ACCESS_ADDRESS_MAX2_43_32		                0x0ac
+#define DMC620_ACCESS_ADDRESS_MIN3_31_00		                0x0b0
+#define DMC620_ACCESS_ADDRESS_MIN3_43_32		                0x0b4
+#define DMC620_ACCESS_ADDRESS_MAX3_31_00		                0x0b8
+#define DMC620_ACCESS_ADDRESS_MAX3_43_32		                0x0bc
+#define DMC620_ACCESS_ADDRESS_MIN4_31_00		                0x0c0
+#define DMC620_ACCESS_ADDRESS_MIN4_43_32		                0x0c4
+#define DMC620_ACCESS_ADDRESS_MAX4_31_00		                0x0c8
+#define DMC620_ACCESS_ADDRESS_MAX4_43_32		                0x0cc
+#define DMC620_ACCESS_ADDRESS_MIN5_31_00		                0x0d0
+#define DMC620_ACCESS_ADDRESS_MIN5_43_32		                0x0d4
+#define DMC620_ACCESS_ADDRESS_MAX5_31_00		                0x0d8
+#define DMC620_ACCESS_ADDRESS_MAX5_43_32		                0x0dc
+#define DMC620_ACCESS_ADDRESS_MIN6_31_00		                0x0e0
+#define DMC620_ACCESS_ADDRESS_MIN6_43_32		                0x0e4
+#define DMC620_ACCESS_ADDRESS_MAX6_31_00		                0x0e8
+#define DMC620_ACCESS_ADDRESS_MAX6_43_32		                0x0ec
+#define DMC620_ACCESS_ADDRESS_MIN7_31_00		                0x0f0
+#define DMC620_ACCESS_ADDRESS_MIN7_43_32		                0x0f4
+#define DMC620_ACCESS_ADDRESS_MAX7_31_00		                0x0f8
+#define DMC620_ACCESS_ADDRESS_MAX7_43_32		                0x0fc
+#define DMC620_DIRECT_ADDR				                          0x108
+#define DMC620_DIRECT_CMD				                            0x10c
+#define DMC620_DCI_REPLAY_TYPE				                      0x110
+#define DMC620_DCI_STRB					                            0x118
+#define DMC620_DCI_DATA					                            0x11c
+#define DMC620_REFRESH_CONTROL				                      0x120
+#define DMC620_MEMORY_TYPE				                          0x128
+#define DMC620_FEATURE_CONFIG				                        0x130
+#define DMC620_SCRUB_CONTROL0				                        0x170
+#define DMC620_SCRUB_CONTROL1				                        0x180
+#define DMC620_SCRUB_CONTROL2				                        0x190
+#define DMC620_SCRUB_CONTROL3				                        0x1a0
+#define DMC620_SCRUB_CONTROL4				                        0x1b0
+#define DMC620_SCRUB_CONTROL5				                        0x1c0
+#define DMC620_SCRUB_CONTROL6				                        0x1d0
+#define DMC620_SCRUB_CONTROL7				                        0x1e0
+#define DMC620_FEATURE_CONTROL				                      0x1f0
+#define DMC620_MUX_CONTROL				                          0x1f4
+#define DMC620_RANK_REMAP_CONTROL			                      0x1f8
+#define DMC620_SCRUB_CONTROL				                        0x1fc
+#define DMC620_T_REFI					                              0x200
+#define DMC620_T_RFC					                              0x204
+#define DMC620_T_MRR					                              0x208
+#define DMC620_T_MRW					                              0x20c
+#define DMC620_T_REFRESH_ENABLE                 				    0x210
+#define DMC620_T_RCD					                              0x218
+#define DMC620_T_RAS					                              0x21c
+#define DMC620_T_RP					                                0x220
+#define DMC620_T_RPALL					                            0x224
+#define DMC620_T_RRD					                              0x228
+#define DMC620_T_ACT_WINDOW			                      	    0x22c
+#define DMC620_T_RTR					                              0x234
+#define DMC620_T_RTW					                              0x238
+#define DMC620_T_RTP					                              0x23c
+#define DMC620_T_WR					                                0x244
+#define DMC620_T_WTR					                              0x248
+#define DMC620_T_WTW					                              0x24c
+#define DMC620_T_XMPD					                              0x254
+#define DMC620_T_EP					                                0x258
+#define DMC620_T_XP					                                0x25c
+#define DMC620_T_ESR					                              0x260
+#define DMC620_T_XSR					                              0x264
+#define DMC620_T_ESRCK					                            0x268
+#define DMC620_T_CKXSR					                            0x26c
+#define DMC620_T_CMD					                              0x270
+#define DMC620_T_PARITY					                            0x274
+#define DMC620_T_ZQCS					                              0x278
+#define DMC620_T_RW_ODT_CLR				                          0x27c
+#define DMC620_T_RDDATA_EN				                          0x300
+#define DMC620_T_PHYRDLAT				                            0x304
+#define DMC620_T_PHYWRLAT				                            0x308
+#define DMC620_RDLVL_CONTROL				                        0x310
+#define DMC620_RDLVL_MRS				                            0x314
+#define DMC620_T_RDLVL_EN				                            0x318
+#define DMC620_T_RDLVL_RR				                            0x31c
+#define DMC620_WDLVL_CONTROL				                        0x320
+#define DMC620_WRLVL_MRS				                            0x324
+#define DMC620_T_WRLVL_EN				                            0x328
+#define DMC620_T_WRLVL_WW				                            0x32c
+#define DMC620_PHY_POWER_CONTROL		                   	    0x348
+#define DMC620_T_LPRESP					                            0x34c
+#define DMC620_PHY_UPDATE_CONTROL			                      0x350
+#define DMC620_ODTH					                                0x354
+#define DMC620_ODT_TIMING				                            0x358
+#define DMC620_ODT_WR_CONTROL_31_00			                    0x360
+#define DMC620_ODT_WR_CONTROL_63_32			                    0x364
+#define DMC620_ODT_RD_CONTROL_31_00			                    0x368
+#define DMC620_ODT_RD_CONTROL_63_32			                    0x36C
 
- #define      MRCTRL0 0x0010
- #define      RFSHTMG 0x0064
- #define       ZQCTL0 0x0180
- #define      HWLPCTL 0x0038
- #define       ODTCFG 0x0240
- #define       ODTMAP 0x0244
- #define        SCHED 0x0250
- #define       SCHED1 0x0254
- #define 	   SCHED3 0x0270
- #define 	   SCHED4 0x0274
 
- #define     PERFHPR1 0x025c
- #define     PERFLPR1 0x0264
- #define      PERFWR1 0x026c
- #define         DBG0 0x0300
- #define       DBGCMD 0x030c
- #define      PCFGR_0 0x0404
- #define      PCFGW_0 0x0408
- #define      PCTRL_0 0x0490
- #define      PCTRL_1 0x0540
- #define      PCTRL_2 0x05f0
- #define      PCTRL_3 0x06a0
- #define 	  SWCTL   0x0320
- #define 	 RFSHCTL0 0x0050
- #define 	 RFSHCTL1 0x0054
+//Added FGPA PHY register map
+#define FPGAPHY_GENERAL_SETUP                               0x0000
+#define FPGAPHY_PLAYBACK_ENABLE                             0x0004
+#define FPGAPHY_PLAYBACK_STATUS                             0x0008
+#define FPGAPHY_WRITE_PIPE_DELAY                            0x000C
+#define FPGAPHY_CA_PIPE_DELAY                               0x0010
+#define FPGAPHY_DATA_BYTE_SEL                               0x0014
+#define FPGAPHY_CTRLUPD_CONFIG                              0x0018
+#define FPGAPHY_DEBUG                                       0x001C
+#define FPGAPHY_INTERRUPT_MASK                              0x0020
+#define FPGAPHY_INTERRUPT_ENABLE                            0x0024
+#define FPGAPHY_INTERRUPT_STATUS                            0x0028
+#define FPGAPHY_INTERRUPT_CLEAR                             0x002C
+#define FPGAPHY_DFICLK_READ_DELAY_RANK_0		                0x0030
+#define FPGAPHY_FCLK_READ_DELAY_RANK_0_BYTES_7_0            0x0034
+#define FPGAPHY_FCLK_READ_DELAY_RANK_0_BYTE_8	            	0x0038
+#define FPGAPHY_DFICLK_READ_DELAY_RANK_1		                0x0040
+#define FPGAPHY_FCLK_READ_DELAY_RANK_1_BYTES_7_0            0x0044
+#define FPGAPHY_FCLK_READ_DELAY_RANK_1_BYTE_8		            0x0048
+#define FPGAPHY_FCLK_READ_OFFSET_RANK_0_BYTES_7_0           0x0050
+#define FPGAPHY_FCLK_READ_OFFSET_RANK_0_BYTE_8	          	0x0054
+#define FPGAPHY_FCLK_READ_OFFSET_RANK_1_BYTES_7_0           0x0058
+#define FPGAPHY_FCLK_READ_OFFSET_RANK_1_BYTE_8              0x005C
+#define FPGAPHY_AUTOTRAIN_LOOP_ADDR		                   		0x0060
+#define FPGAPHY_TRAIN_STATUS						                    0x0064
+#define FPGAPHY_READ_TRAINING_CAPTURE_0	                		0x0068
+#define FPGAPHY_READ_TRAINING_CAPTURE_1	                		0x006C
+#define FPGAPHY_REVISION_REGISTER				                    0x0070
+#define FPGAPHY_READ_PHASE_DELAY_RANK_0		                 	0x0074
+#define FPGAPHY_READ_PHASE_DELAY_RANK_1		                 	0x0078
+#define PLAYBACK_ENGINE_CODE_0				                   		0x0800
+#define PLAYBACK_ENGINE_CODE_1				                   		0x0804
+#define PLAYBACK_ENGINE_CODE_2				                   		0x0808
+#define PLAYBACK_ENGINE_CODE_3				                   		0x080C
+#define PLAYBACK_ENGINE_CODE_4				                   		0x0810
+#define PLAYBACK_ENGINE_CODE_8				                   		0x0820
+#define PLAYBACK_ENGINE_CODE_9				                   		0x0824
+#define PLAYBACK_ENGINE_CODE_10				                   		0x0828
+#define PLAYBACK_ENGINE_CODE_11				                   		0x082c
+#define PLAYBACK_ENGINE_CODE_12				                   		0x0830
+#define PLAYBACK_ENGINE_CODE_16				                   		0x0840
+#define PLAYBACK_ENGINE_CODE_17				                   		0x0844
+#define PLAYBACK_ENGINE_CODE_18				                   		0x0848
+#define PLAYBACK_ENGINE_CODE_19				                   		0x084c
+#define PLAYBACK_ENGINE_CODE_20				                   		0x0850
+#define PLAYBACK_ENGINE_CODE_24				                   		0x0860
+#define PLAYBACK_ENGINE_CODE_25				                   		0x0864
+#define PLAYBACK_ENGINE_CODE_26				                   		0x0868
+#define PLAYBACK_ENGINE_CODE_27				                   		0x086c
+#define PLAYBACK_ENGINE_CODE_28				                   		0x0870
+#define PLAYBACK_ENGINE_CODE_32				                   		0x0880
+#define PLAYBACK_ENGINE_CODE_33				                   		0x0884
+#define PLAYBACK_ENGINE_CODE_34				                   		0x0888
+#define PLAYBACK_ENGINE_CODE_35				                   		0x088c
+#define PLAYBACK_ENGINE_CODE_36				                   		0x0890
+#define PLAYBACK_ENGINE_CODE_40				                   		0x08a0
+#define PLAYBACK_ENGINE_CODE_41				                   		0x08a4
+#define PLAYBACK_ENGINE_CODE_42				                   		0x08a8
+#define PLAYBACK_ENGINE_CODE_43				                   		0x08ac
+#define PLAYBACK_ENGINE_CODE_44				                   		0x08b0
+#define PLAYBACK_ENGINE_CODE_48				                   		0x08c0
+#define PLAYBACK_ENGINE_CODE_49				                   		0x08c4
+#define PLAYBACK_ENGINE_CODE_50				                   		0x08c8
+#define PLAYBACK_ENGINE_CODE_51				                   		0x08cc
+#define PLAYBACK_ENGINE_CODE_52				                   		0x08d0
+#define PLAYBACK_ENGINE_CODE_56				                   		0x08e0
+#define PLAYBACK_ENGINE_CODE_57				                   		0x08e4
+#define PLAYBACK_ENGINE_CODE_58				                   		0x08e8
+#define PLAYBACK_ENGINE_CODE_59				                   		0x08ec
+#define PLAYBACK_ENGINE_CODE_60				                   		0x08f0
+#define PLAYBACK_ENGINE_CODE_64				                   		0x0900
+#define PLAYBACK_ENGINE_CODE_65				                   		0x0904
+#define PLAYBACK_ENGINE_CODE_66				                   		0x0908
+#define PLAYBACK_ENGINE_CODE_67				                   		0x090c
+#define PLAYBACK_ENGINE_CODE_68				                   		0x0910
+#define PLAYBACK_ENGINE_CODE_72				                   		0x0920
+#define PLAYBACK_ENGINE_CODE_73				                   		0x0924
+#define PLAYBACK_ENGINE_CODE_74				                   		0x0928
+#define PLAYBACK_ENGINE_CODE_75				                   		0x092c
+#define PLAYBACK_ENGINE_CODE_76				                   		0x0930
+#define PLAYBACK_ENGINE_CODE_80				                   		0x0940
+#define PLAYBACK_ENGINE_CODE_81				                   		0x0944
+#define PLAYBACK_ENGINE_CODE_82				                   		0x0948
+#define PLAYBACK_ENGINE_CODE_83				                   		0x094c
+#define PLAYBACK_ENGINE_CODE_84				                   		0x0950
+#define PLAYBACK_ENGINE_CODE_88				                   		0x0960
+#define PLAYBACK_ENGINE_CODE_89				                   		0x0964
+#define PLAYBACK_ENGINE_CODE_90				                   		0x0968
+#define PLAYBACK_ENGINE_CODE_91				                   		0x096c
+#define PLAYBACK_ENGINE_CODE_92				                   		0x0970
+#define PLAYBACK_ENGINE_CODE_96				                   		0x0980
+#define PLAYBACK_ENGINE_CODE_97				                   		0x0984
+#define PLAYBACK_ENGINE_CODE_98				                   		0x0988
+#define PLAYBACK_ENGINE_CODE_99				                   		0x098c
+#define PLAYBACK_ENGINE_CODE_100			                   		0x0990
+#define PLAYBACK_ENGINE_CODE_104			                   		0x09a0
+#define PLAYBACK_ENGINE_CODE_105			                   		0x09a4
+#define PLAYBACK_ENGINE_CODE_106			                   		0x09a8
+#define PLAYBACK_ENGINE_CODE_107			                   		0x09ac
+#define PLAYBACK_ENGINE_CODE_108			                   		0x09b0
+#define PLAYBACK_ENGINE_CODE_112			                   		0x09c0
+#define PLAYBACK_ENGINE_CODE_113			                   		0x09c4
+#define PLAYBACK_ENGINE_CODE_114			                   		0x09c8
+#define PLAYBACK_ENGINE_CODE_115			                   		0x09cc
+#define PLAYBACK_ENGINE_CODE_116			                   		0x09d0
+#define PLAYBACK_ENGINE_CODE_120			                   		0x09e0
+#define PLAYBACK_ENGINE_CODE_121			                   		0x09e4
+#define PLAYBACK_ENGINE_CODE_122			                   		0x09e8
+#define PLAYBACK_ENGINE_CODE_123			                   		0x09ec
+#define PLAYBACK_ENGINE_CODE_124			                   		0x09f0
+#define PLAYBACK_ENGINE_CODE_128			                   		0x0a00
+#define PLAYBACK_ENGINE_CODE_129			                   		0x0a04
+#define PLAYBACK_ENGINE_CODE_130			                   		0x0a08
+#define PLAYBACK_ENGINE_CODE_131			                   		0x0a0c
+#define PLAYBACK_ENGINE_CODE_132			                   		0x0a10
+#define PLAYBACK_ENGINE_CODE_136			                   		0x0a20
+#define PLAYBACK_ENGINE_CODE_137			                   		0x0a24
+#define PLAYBACK_ENGINE_CODE_138			                   		0x0a28
+#define PLAYBACK_ENGINE_CODE_139			                   		0x0a2c
+#define PLAYBACK_ENGINE_CODE_140			                   		0x0a30
+#define PLAYBACK_ENGINE_CODE_144			                   		0x0a40
+#define PLAYBACK_ENGINE_CODE_145			                   		0x0a44
+#define PLAYBACK_ENGINE_CODE_146			                   		0x0a48
+#define PLAYBACK_ENGINE_CODE_147			                   		0x0a4c
+#define PLAYBACK_ENGINE_CODE_148			                   		0x0a50
+#define PLAYBACK_ENGINE_CODE_152			                   		0x0a60
+#define PLAYBACK_ENGINE_CODE_153			                   		0x0a64
+#define PLAYBACK_ENGINE_CODE_154			                   		0x0a68
+#define PLAYBACK_ENGINE_CODE_155			                   		0x0a6c
+#define PLAYBACK_ENGINE_CODE_156			                   		0x0a70
+#define PLAYBACK_ENGINE_CODE_160			                   		0x0a80
+#define PLAYBACK_ENGINE_CODE_161			                   		0x0a84
+#define PLAYBACK_ENGINE_CODE_162			                   		0x0a88
+#define PLAYBACK_ENGINE_CODE_163			                   		0x0a8c
+#define PLAYBACK_ENGINE_CODE_164			                   		0x0a90
+#define PLAYBACK_ENGINE_CODE_168			                   		0x0aa0
+#define PLAYBACK_ENGINE_CODE_169			                   		0x0aa4
+#define PLAYBACK_ENGINE_CODE_170			                   		0x0aa8
+#define PLAYBACK_ENGINE_CODE_171			                   		0x0aac
+#define PLAYBACK_ENGINE_CODE_172			                   		0x0ab0
+#define PLAYBACK_ENGINE_CODE_176			                   		0x0ac0
+#define PLAYBACK_ENGINE_CODE_177			                   		0x0ac4
+#define PLAYBACK_ENGINE_CODE_178			                   		0x0ac8
+#define PLAYBACK_ENGINE_CODE_179			                   		0x0acc
+#define PLAYBACK_ENGINE_CODE_180			                   		0x0ad0
+#define PLAYBACK_ENGINE_CODE_184			                   		0x0ae0
+#define PLAYBACK_ENGINE_CODE_185			                   		0x0ae4
+#define PLAYBACK_ENGINE_CODE_186			                   		0x0ae8
+#define PLAYBACK_ENGINE_CODE_187			                   		0x0aec
+#define PLAYBACK_ENGINE_CODE_188			                   		0x0af0
+#define PLAYBACK_ENGINE_CODE_192			                   		0x0b00
+#define PLAYBACK_ENGINE_CODE_193			                   		0x0b04
+#define PLAYBACK_ENGINE_CODE_194			                   		0x0b08
+#define PLAYBACK_ENGINE_CODE_195			                   		0x0b0c
+#define PLAYBACK_ENGINE_CODE_196			                   		0x0b10
+#define PLAYBACK_ENGINE_CODE_200			                   		0x0b20
+#define PLAYBACK_ENGINE_CODE_201			                   		0x0b24
+#define PLAYBACK_ENGINE_CODE_202			                   		0x0b28
+#define PLAYBACK_ENGINE_CODE_203			                   		0x0b2c
+#define PLAYBACK_ENGINE_CODE_204			                   		0x0b30
+#define PLAYBACK_ENGINE_CODE_208			                   		0x0b40
+#define PLAYBACK_ENGINE_CODE_209			                   		0x0b44
+#define PLAYBACK_ENGINE_CODE_210			                   		0x0b48
+#define PLAYBACK_ENGINE_CODE_211			                   		0x0b4c
+#define PLAYBACK_ENGINE_CODE_212			                   		0x0b50
+#define PLAYBACK_ENGINE_CODE_216			                   		0x0b60
+#define PLAYBACK_ENGINE_CODE_217			                   		0x0b64
+#define PLAYBACK_ENGINE_CODE_218			                   		0x0b68
+#define PLAYBACK_ENGINE_CODE_219			                   		0x0b6c
+#define PLAYBACK_ENGINE_CODE_220			                   		0x0b70
+#define PLAYBACK_ENGINE_CODE_224			                   		0x0b80
+#define PLAYBACK_ENGINE_CODE_225			                   		0x0b84
+#define PLAYBACK_ENGINE_CODE_226			                   		0x0b88
+#define PLAYBACK_ENGINE_CODE_227			                   		0x0b8c
+#define PLAYBACK_ENGINE_CODE_228			                   		0x0b90
+#define PLAYBACK_ENGINE_CODE_232			                   		0x0ba0
+#define PLAYBACK_ENGINE_CODE_233			                   		0x0ba4
+#define PLAYBACK_ENGINE_CODE_234			                   		0x0ba8
+#define PLAYBACK_ENGINE_CODE_235			                   		0x0bac
+#define PLAYBACK_ENGINE_CODE_236			                   		0x0bb0
+#define PLAYBACK_ENGINE_CODE_240			                   		0x0bc0
+#define PLAYBACK_ENGINE_CODE_241			                   		0x0bc4
+#define PLAYBACK_ENGINE_CODE_242			                   		0x0bc8
+#define PLAYBACK_ENGINE_CODE_243			                   		0x0bcc
+#define PLAYBACK_ENGINE_CODE_244			                   		0x0bd0
+#define PLAYBACK_ENGINE_CODE_248			                   		0x0be0
+#define PLAYBACK_ENGINE_CODE_249			                   		0x0be4
+#define PLAYBACK_ENGINE_CODE_250			                   		0x0be8
+#define PLAYBACK_ENGINE_CODE_251			                   		0x0bec
+#define PLAYBACK_ENGINE_CODE_252			                   		0x0bf0
+#define PLAYBACK_ENGINE_CODE_256			                   		0x0c00
+#define PLAYBACK_ENGINE_CODE_257			                   		0x0c04
+#define PLAYBACK_ENGINE_CODE_258			                   		0x0c08
+#define PLAYBACK_ENGINE_CODE_259			                   		0x0c0c
+#define PLAYBACK_ENGINE_CODE_260			                   		0x0c10
+#define PLAYBACK_ENGINE_CODE_264			                   		0x0c20
+#define PLAYBACK_ENGINE_CODE_265			                   		0x0c24
+#define PLAYBACK_ENGINE_CODE_266			                   		0x0c28
+#define PLAYBACK_ENGINE_CODE_267			                   		0x0c2c
+#define PLAYBACK_ENGINE_CODE_268			                   		0x0c30
+//
 
-/* Register definitions for PHY */
-#define DMC620_PHY_INIT_COMPLETE      0x000
-#define DMC620_PHY_READ_DELAY         0x008
-#define DMC620_PHY_CAPTURE_MASK       0x00C
-#define DMC620_PHY_T_CONTROL_DELAY    0x014
-#define DMC620_PHY_WR_LATENCY         0x018
-#define DMC620_PHY_RDDATA_EN          0x01c
-#define DMC620_PHY_RD_LATENCY         0x020
-#define DMC620_PHY_CLP_ACK            0x030
-#define DMC620_PHY_RDLVL_REQ          0x044
-#define DMC620_PHY_WRLVL_REQ          0x04C
-#define DMC620_PHY_WR_PREMBL          0x058
-#define DMC620_PHY_DELAY_SEL	      0x820
-#define DMC620_PHY_REF_EN	      0x824
-#define DMC620_PHY_T_CTRL_UPD_MIN     0x828
-#define DMC620_PHY_OP_DELAY1          0x830
-#define DMC620_PHY_OP_DELAY2          0x834
-#define DMC620_PHY_OP_DELAY3          0x838
-#define DMC620_PHY_OP_DELAY4          0x83C
+#define DFIMISC                                             0x01b0
+#define DFITMG0                                             0x0190
+#define DFITMG1                                             0x0194
+#define DFILPCFG0                                           0x0198
+#define DFIUPD0                                             0x01a0
+#define DFIUPD1                                             0x01a4
+#define DFIUPD2                                             0x01a8
+#define STAT	                                              0x0004
+#define SWSTAT	                                            0x0324
 
-#define DMC620_STATUS  	      				    0x000
-#define DMC620_CONFIG  	      				    0x004
-#define DMC620_COMMAND 	      	  	     		    0x008
-#define DMC620_ADD_CONTROL	      		   	    0x010
-#define DMC620_DEC_CONTROL	      		   	    0x014
-#define DMC620_FORMAT_CONTROL	      		   	    0x018
-#define DMC620_ADDRESS_MAP	      		   	    0x01c
-#define DMC620_LOW_PWR_CONTROL	      		   	    0x020
-#define DMC620_TURNAROUND_CONTROL     		   	    0x028
-#define DMC620_HIT_TURNAROUND_CONTROL 		   	    0x02c
-#define DMC620_QOS_CLASS_CONTROL      		   	    0x030
-#define DMC620_ESCALATION_CONTROL     		   	    0x034
-#define DMC620_QV_CONTROL_31_00       		   	    0x038
-#define DMC620_QV_CONTROL_63_32       		   	    0x03c
-#define DMC620_RT_CONTROL_31_00       		   	    0x040
-#define DMC620_RT_CONTROL_63_32       		   	    0x044
-#define DMC620_TIMEOUT_CONTROL        		   	    0x048
-#define DMC620_CREDIT_CONTROL        		   	    0x04c
-#define DMC620_WRITE_PRIORITY_CONTROL_31_00        	    0x050
-#define DMC620_WRITE_PRIORITY_CONTROL_63_32        	    0x054
-#define DMC620_QUEUE_THRESHOLD_CONTROL_31_00                0x058
-#define DMC620_QUEUE_THRESHOLD_CONTROL_63_32                0x05c
-#define DMC620_ADDR_SHUTTER_31_00                     	    0x060
-#define DMC620_ADDR_SHUTTER_63_32                     	    0x064
-#define DMC620_ADDR_SHUTTER_95_64                     	    0x068
-#define DMC620_ADDR_SHUTTER_127_96                     	    0x06c
-#define DMC620_ADDR_SHUTTER_159_128                    	    0x070
-#define DMC620_ADDR_SHUTTER_191_160                    	    0x074
-#define DMC620_MEMORY_ADDRESS_MAX_31_00    	      	    0x078
-#define DMC620_MEMORY_ADDRESS_MAX_43_32			    0x07C
-#define DMC620_ACCESS_ADDRESS_MIN0_31_00   		    0x080
-#define DMC620_ACCESS_ADDRESS_MIN0_43_32		    0x084
-#define DMC620_ACCESS_ADDRESS_MAX0_31_00   		    0x088
-#define DMC620_ACCESS_ADDRESS_MAX0_43_32   		    0x08C
-#define DMC620_ACCESS_ADDRESS_MIN1_31_00   		    0x090
-#define DMC620_ACCESS_ADDRESS_MIN1_43_32   		    0x094
-#define DMC620_ACCESS_ADDRESS_MAX1_31_00   		    0x098
-#define DMC620_ACCESS_ADDRESS_MAX1_43_32   		    0x09C
-#define DMC620_ACCESS_ADDRESS_MIN2_31_00   		    0x0a0
-#define DMC620_ACCESS_ADDRESS_MIN2_43_32   		    0x0a4
-#define DMC620_ACCESS_ADDRESS_MAX2_31_00   		    0x0a8
-#define DMC620_ACCESS_ADDRESS_MAX2_43_32   		    0x0ac
-#define DMC620_ACCESS_ADDRESS_MIN3_31_00   		    0x0b0
-#define DMC620_ACCESS_ADDRESS_MIN3_43_32   		    0x0b4
-#define DMC620_ACCESS_ADDRESS_MAX3_31_00   		    0x0b8
-#define DMC620_ACCESS_ADDRESS_MAX3_43_32   		    0x0bc
-#define DMC620_ACCESS_ADDRESS_MIN4_31_00   		    0x0c0
-#define DMC620_ACCESS_ADDRESS_MIN4_43_32   		    0x0c4
-#define DMC620_ACCESS_ADDRESS_MAX4_31_00   		    0x0c8
-#define DMC620_ACCESS_ADDRESS_MAX4_43_32   		    0x0cc
-#define DMC620_ACCESS_ADDRESS_MIN5_31_00   		    0x0d0
-#define DMC620_ACCESS_ADDRESS_MIN5_43_32   		    0x0d4
-#define DMC620_ACCESS_ADDRESS_MAX5_31_00   		    0x0d8
-#define DMC620_ACCESS_ADDRESS_MAX5_43_32   		    0x0dc
-#define DMC620_ACCESS_ADDRESS_MIN6_31_00   		    0x0e0
-#define DMC620_ACCESS_ADDRESS_MIN6_43_32   		    0x0e4
-#define DMC620_ACCESS_ADDRESS_MAX6_31_00   		    0x0e8
-#define DMC620_ACCESS_ADDRESS_MAX6_43_32   		    0x0ec
-#define DMC620_ACCESS_ADDRESS_MIN7_31_00   		    0x0f0
-#define DMC620_ACCESS_ADDRESS_MIN7_43_32   		    0x0f4
-#define DMC620_ACCESS_ADDRESS_MAX7_31_00   		    0x0f8
-#define DMC620_ACCESS_ADDRESS_MAX7_43_32   		    0x0fc
-#define DMC620_DIRECT_ADDR	      			    0x108
-#define DMC620_DIRECT_CMD	      			    0x10c
-#define DMC620_DCI_REPLAY_TYPE	      			    0x110
-#define DMC620_DCI_STRB      	      			    0x118
-#define DMC620_DCI_DATA      	      			    0x11c
-#define DMC620_REFRESH_CONTROL	      			    0x120
-#define DMC620_MEMORY_TYPE	      			    0x128
-#define DMC620_FEATURE_CONFIG         			    0x130
-#define DMC620_SCRUB_CONTROL0	      			    0x170
-#define DMC620_SCRUB_CONTROL1	      			    0x180
-#define DMC620_SCRUB_CONTROL2	      			    0x190
-#define DMC620_SCRUB_CONTROL3	      			    0x1a0
-#define DMC620_SCRUB_CONTROL4	      			    0x1b0
-#define DMC620_SCRUB_CONTROL5	      			    0x1c0
-#define DMC620_SCRUB_CONTROL6	      			    0x1d0
-#define DMC620_SCRUB_CONTROL7	      			    0x1e0
-#define DMC620_FEATURE_CONTROL          		    0x1f0
-#define DMC620_MUX_CONTROL              		    0x1f4
-#define DMC620_RANK_REMAP_CONTROL       		    0x1f8
-#define DMC620_SCRUB_CONTROL    	      	      	    0x1fc
-#define DMC620_T_REFI  	      	      	      	      	    0x200
-#define DMC620_T_RFC		      			    0x204
-#define DMC620_T_MRR		      			    0x208
-#define DMC620_T_MRW		      			    0x20c
-#define DMC620_T_REFRESH_ENABLE	      	      	      	    0x210
-#define DMC620_T_RCD		      			    0x218
-#define DMC620_T_RAS		      			    0x21c
-#define DMC620_T_RP		      			    0x220
-#define DMC620_T_RPALL 	      	      			    0x224
-#define DMC620_T_RRD		      			    0x228
-#define DMC620_T_ACT_WINDOW	      			    0x22c
-#define DMC620_T_RTR		      			    0x234
-#define DMC620_T_RTW		      			    0x238
-#define DMC620_T_RTP		      			    0x23c
-#define DMC620_T_WR		      			    0x244
-#define DMC620_T_WTR		      			    0x248
-#define DMC620_T_WTW		      			    0x24c
-#define DMC620_T_XMPD		      			    0x254
-#define DMC620_T_EP		      			    0x258
-#define DMC620_T_XP		      			    0x25c
-#define DMC620_T_ESR		      			    0x260
-#define DMC620_T_XSR		      			    0x264
-#define DMC620_T_ESRCK 	      	      			    0x268
-#define DMC620_T_CKXSR 	      	      			    0x26c
-#define DMC620_T_CMD		      			    0x270
-#define DMC620_T_PARITY	      	      			    0x274
-#define DMC620_T_ZQCS	      	      			    0x278
-#define DMC620_T_RW_ODT_CLR    	      			    0x27c
-#define DMC620_T_RDDATA_EN	      			    0x300
-#define DMC620_T_PHYRDLAT	      			    0x304
-#define DMC620_T_PHYWRLAT	      			    0x308
-#define DMC620_RDLVL_CONTROL	      			    0x310
-#define DMC620_RDLVL_MRS	      			    0x314
-#define DMC620_T_RDLVL_EN	      			    0x318
-#define DMC620_T_RDLVL_RR	      			    0x31c
-#define DMC620_WDLVL_CONTROL	      			    0x320
-#define DMC620_WRLVL_MRS	      			    0x324
-#define DMC620_T_WRLVL_EN	      			    0x328
-#define DMC620_T_WRLVL_WW	      			    0x32c
-#define DMC620_PHY_POWER_CONTROL      			    0x348
-#define DMC620_T_LPRESP	      	      			    0x34c
-#define DMC620_PHY_UPDATE_CONTROL     			    0x350
-#define DMC620_ODTH   		      			    0x354
-#define DMC620_ODT_TIMING	      			    0x358
-#define DMC620_ODT_WR_CONTROL_31_00   			    0x360
-#define DMC620_ODT_WR_CONTROL_63_32   			    0x364
-#define DMC620_ODT_RD_CONTROL_31_00   			    0x368
-#define DMC620_ODT_RD_CONTROL_63_32   			    0x36C
+#define DBG1	                                              0x0304
+#define PWRCTL		                                          0x0030
+#define MSTR	                                              0x0000
+#define	MRSTAT                                              0x0018
+#define INIT0                                               0x00d0
+#define INIT1                                               0x00d4
+#define INIT2                                               0x00d8
+#define INIT3                                               0x00dc
+#define INIT4                                               0x00e0
+#define INIT5                                               0x00e4
+#define INIT6                                               0x00e8
+#define	RANKCTL		                                          0x00f4
+#define DRAMTMG0                                          	0x0100
+#define DRAMTMG1                                          	0x0104
+#define DRAMTMG2                                          	0x0108
+#define DRAMTMG3                                          	0x010c
+#define DRAMTMG4                                          	0x0110
+#define DRAMTMG5                                          	0x0114
+#define DRAMTMG6                                          	0x0118
+#define DRAMTMG7                                          	0x011c
+#define DRAMTMG13                                           0x0134
+#define DRAMTMG14                                           0x0138
+#define ADDRMAP0                                            0x0200
+#define ADDRMAP1                                           	0x0204
+#define ADDRMAP2                                           	0x0208
+#define ADDRMAP3                                           	0x020c
+#define ADDRMAP4                                           	0x0210
+#define ADDRMAP5                                           	0x0214
+#define ADDRMAP6                                           	0x0218
+#define ADDRMAP7                                          	0x021c
+#define ADDRMAP8                                          	0x0220
+#define ADDRMAP9                                          	0x0224
+#define ADDRMAP10                                          	0x0228
+#define ADDRMAP11                                          	0x022c
 
+#define MRCTRL0                                             0x0010
+#define RFSHTMG                                             0x0064
+#define ZQCTL0                                              0x0180
+#define HWLPCTL                                             0x0038
+#define ODTCFG                                              0x0240
+#define ODTMAP                                              0x0244
+#define SCHED                                               0x0250
+#define SCHED1                                              0x0254
+#define SCHED3                                              0x0270
+#define SCHED4                                              0x0274
 
+#define PERFHPR1                                            0x025c
+#define PERFLPR1                                            0x0264
+#define PERFWR1                                             0x026c
+#define DBG0                                                0x0300
+#define DBGCMD                                              0x030c
+#define PCFGR_0                                             0x0404
+#define PCFGW_0                                             0x0408
+#define PCTRL_0                                             0x0490
+#define PCTRL_1                                             0x0540
+#define PCTRL_2                                             0x05f0
+#define PCTRL_3                                             0x06a0
+#define	SWCTL                                               0x0320
+#define	RFSHCTL0                                            0x0050
+#define	RFSHCTL1                                            0x0054
 
+#endif
diff --git a/module/snps_umctl/include/snps_defines.h b/module/snps_umctl/include/snps_defines.h
index 940cb833..43ac5b52 100644
--- a/module/snps_umctl/include/snps_defines.h
+++ b/module/snps_umctl/include/snps_defines.h
@@ -83,6 +83,16 @@
 #ifndef UMCTL_DEFINES_H
 #define UMCTL_DEFINES_H
 
+#ifndef BA_BEFORE_BG
+#define BA_BEFORE_BG 1
+#endif
+#ifndef DMC_WCK_ALWAYS_ON
+#define DMC_WCK_ALWAYS_ON 1
+#endif
+#ifndef QOS_DISABLE
+#define QOS_DISABLE 0
+#endif
+
 #ifndef UMCTL_RDWR_IDLE_GAP
 #define UMCTL_RDWR_IDLE_GAP 0
 #endif
diff --git a/module/snps_umctl/include/test_defs.h b/module/snps_umctl/include/test_defs.h
index f5613d76..2e94ab8f 100644
--- a/module/snps_umctl/include/test_defs.h
+++ b/module/snps_umctl/include/test_defs.h
@@ -4,6 +4,7 @@
 
 #define HCG 1       // Enable High Level Clock Gating
 
+
 //Clock Divider Settings
 #define CCI_NO_SCP
 
@@ -51,7 +52,13 @@
 // DMC500 settings
 #define DMC_BASE_ADDR       0xC1000000
 #define DMC_OFFSET          0x01000000
+
+#ifndef ROS_COMPONENT
 #define DMC_PHY_BASE_ADDR   0xA8000000
+#else
+#define DMC_PHY_BASE_ADDR   0x9Fb60000
+#endif
+
 #define DMC_PHY_OFFSET      0x00010000
 #define UMCTL2_INIT_FUN     "umctl2_lpddr4_3200.h"
 #define MULTI_PORT
@@ -77,13 +84,16 @@
 #define DMC_BANK_GROUP 0       // 0 bank groups
 #define MEM_CHANNEL 4
 #define DMC_ADDR_SHUTTER 2
+#define DMC_STRIPE  256
+#define DMC_BANK_STRIPE  256
 
 #ifndef DMC_HASH_ENABLE
 #define DMC_HASH_ENABLE 1
 #endif
 
+
 #define DMC_ADDR_DEC 1 //rcb
-  
+
 // LPDDR4-4267 x16 (cernan)
 #define LPDDR4_4267
 #define DMC_DEVICE_WIDTH 2 // x16
@@ -102,6 +112,8 @@
 
 #define NPU0_BASE 0x050000000
 
+
+
 #define SDRAM_BASE_ADDR               0x80000000
 #define SDRAM_HIGH_ADDR               0x880000000
 #define SDRAM_END_ADDR                (0x880000000+(size_t)PMA_SIZE*0x40000000)
@@ -115,4 +127,3 @@
 #define CPUFREQ 2700
 
 #endif // TEST_DEFS_H
-
-- 
2.25.1

